File: eco-core/core-backend/src/main/java/com/willfp/eco/internal/gui/slot/EcoFillerSlot.java
package com.willfp.eco.internal.gui.slot;

import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;

public class EcoFillerSlot extends EcoSlot {
    public EcoFillerSlot(@NotNull final ItemStack itemStack) {
        super((player) -> itemStack, null, null, null, null, null);
    }
}

File: eco-core/core-backend/src/main/kotlin/com/willfp/eco/internal/gui/slot/EcoFillerSlot.kt
package com.willfp.eco.internal.gui.slot

import com.willfp.eco.core.gui.slot.functional.SlotHandler
import org.bukkit.inventory.ItemStack

private val noop = SlotHandler { _, _, _ -> }

class EcoFillerSlot(itemStack: ItemStack) : EcoSlot(
    { _, _ -> itemStack },
    noop,
    noop,
    noop,
    noop,
    noop,
    { _, _, prev -> prev }
)
